ocaml/parsing/jane_syntax_parsing.ml

```diff
@@ -658,6 +659,22 @@ module Expression0 = Make_with_attribute (struct
   let with_attributes expr pexp_attributes = { expr with pexp_attributes }
 end)
 
+(** Cases; embedded using an attribute on the rhs. *)
+module Case0 = Make_with_attribute (struct
```

Comment:
This language extension is not "erasable" &mdash; we're not easily going to be able to rewrite programs that use this feature into ones that don't simply by dropping attributes. For that reason, we have more flexibility with our choice of embedding. In particular, we can use extensions instead of attributes as our means of embedding.

If we do this, an uninterpreted extension node will be wrapped around the expression (instead of just adding attributes) so, if the extension node somehow makes it through compilation, the middle end will raise an exception instead of silently miscompiling.

Reply:
Makes sense! The new encoding applies the extension node to the case rhs, let me know if this doesn't make sense.